tests/integration/testlib/assertlib.py

```diff
@@ -53,3 +54,17 @@ def assert_absent(*ifnames):
 
     current_state = statelib.show_only(ifnames)
     assert not current_state[INTERFACES]
+
+
+def iface_macs(state):
+    for ifstate in state[Interface.KEY]:
+        yield ifstate[Interface.MAC]
+
+
+def assert_mac_address(state, expected_mac=None):
+    """ Asserts that all MAC addresses of ifaces in a state are the same """
+    macs = list(iface_macs(state))
```

Comment:
This generator advantage is lost here.
Using a generator, you do not need to collect all values in a list before you check, you just pass one by one and compare. So lets try not to convert it to a list.

Regarding the expected mac argument, I would just do something like this:
```
macs = iface_macs()
if not expected_mac:
    expected_mac = next(macs) # We fetched the first mac.
# from now on we can continue as usuall
for mac in macs:
    assert expected_mac == mac:
```